osu.Game.Tournament/Screens/Editors/Components/LadderResetTeamsDialog.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Overlays.Dialog;
+using osu.Game.Tournament.Screens.Ladder.Components;
+
+namespace osu.Game.Tournament.Screens.Editors.Components
+{
+    public partial class LadderResetTeamsDialog : DangerousActionDialog
+    {
+        public LadderResetTeamsDialog(Container<DrawableTournamentMatch> matchesContainer)
```

Comment:
Please don't pass containers to these dialogs like this, pass the `Action` to execute instead.

Hell I'm not sure you even need to make all these classes. You could probably use `DangerousActionDialog` on its own and locally set `HeaderText`, `Icon` and `DangerousAction` as you see fit.

Reply:
`DangerousActionDialog` is declared abstract and its constructor as well as DangerousAction are protected. It looks like `DangerousAction` wasn't intended to be used directly and was meant to be subclassed. 
Of course I could remove the `abstract` modifiers and change both ctor and DangerousAction to public, thoughts?

Reply:
I guess I didn't notice _that_, somehow.

Probably keep the classes. Note about passing `Action` still applies, though.